editor/project_manager.cpp

```diff
@@ -62,1118 +63,1121 @@
-		add_child(dialog_error);
+void ProjectListItemControl::set_is_grayed(bool p_grayed) {
+	if (p_grayed) {
+		main_vbox->set_modulate(Color(1, 1, 1, 0.5));
+		// Don't make the icon less prominent if the parent is already grayed out.
+		explore_button->set_modulate(Color(1, 1, 1, 1.0));
+	} else {
+		main_vbox->set_modulate(Color(1, 1, 1, 1.0));
+		explore_button->set_modulate(Color(1, 1, 1, 0.5));
 	}
-};
+}
 
-class ProjectListItemControl : public HBoxContainer {
-	GDCLASS(ProjectListItemControl, HBoxContainer)
-public:
-	TextureButton *favorite_button;
-	TextureRect *icon;
-	bool icon_needs_reload;
-	bool hover;
+void ProjectListItemControl::_favorite_button_pressed() {
+	emit_signal(SNAME("favorite_pressed"));
+}
 
-	ProjectListItemControl() {
-		favorite_button = nullptr;
-		icon = nullptr;
-		icon_needs_reload = true;
-		hover = false;
+void ProjectListItemControl::_explore_button_pressed() {
+	emit_signal(SNAME("explore_pressed"));
+}
 
-		set_focus_mode(FocusMode::FOCUS_ALL);
-	}
+void ProjectListItemControl::_bind_methods() {
+	ADD_SIGNAL(MethodInfo("favorite_pressed"));
+	ADD_SIGNAL(MethodInfo("explore_pressed"));
+}
 
-	void set_is_favorite(bool fav) {
-		favorite_button->set_modulate(fav ? Color(1, 1, 1, 1) : Color(1, 1, 1, 0.2));
-	}
+ProjectListItemControl::ProjectListItemControl() {
+	set_focus_mode(FocusMode::FOCUS_ALL);
 
-	void _notification(int p_what) {
-		switch (p_what) {
-			case NOTIFICATION_MOUSE_ENTER: {
-				hover = true;
-				queue_redraw();
-			} break;
+	VBoxContainer *favorite_box = memnew(VBoxContainer);
+	favorite_box->set_alignment(BoxContainer::ALIGNMENT_CENTER);
+	add_child(favorite_box);
 
-			case NOTIFICATION_MOUSE_EXIT: {
-				hover = false;
-				queue_redraw();
-			} break;
+	favorite_button = memnew(TextureButton);
+	favorite_button->set_name("FavoriteButton");
+	// This makes the project's "hover" style display correctly when hovering the favorite icon.
+	favorite_button->set_mouse_filter(MOUSE_FILTER_PASS);
+	favorite_box->add_child(favorite_button);
+	favorite_button->connect("pressed", callable_mp(this, &ProjectListItemControl::_favorite_button_pressed));
 
-			case NOTIFICATION_DRAW: {
-				if (hover) {
-					draw_style_box(get_theme_stylebox(SNAME("hover"), SNAME("Tree")), Rect2(Point2(), get_size()));
-				}
-			} break;
-		}
+	project_icon = memnew(TextureRect);
+	project_icon->set_name("ProjectIcon");
+	project_icon->set_v_size_flags(SIZE_SHRINK_CENTER);
+	add_child(project_icon);
+
+	main_vbox = memnew(VBoxContainer);
+	main_vbox->set_h_size_flags(Control::SIZE_EXPAND_FILL);
+	add_child(main_vbox);
+
+	Control *ec = memnew(Control);
+	ec->set_custom_minimum_size(Size2(0, 1));
+	ec->set_mouse_filter(MOUSE_FILTER_PASS);
+	main_vbox->add_child(ec);
+
+	// Top half, title and unsupported features labels.
+	{
+		HBoxContainer *title_hb = memnew(HBoxContainer);
+		main_vbox->add_child(title_hb);
+
+		project_title = memnew(Label);
+		project_title->set_name("ProjectName");
```

Comment:
What is the name for?

Reply:
Some nodes in there already had names assigned, and there is that idea that we should start naming nodes in the editor to help with debug and plugins (less relevant in PM). So I adjusted the ones that were there and added a couple next to them, because why not.

I don't think it's terribly important, if you are asking this with the context of your tags PR. So feel free to avoid doing such changes.